### ISSUE TITLE
Build your own hardware wallet using a smartcard with Satochip

### ISSUE THREAD

**Issue by Toporin** [TABConf/2023.tabconf.com#62]
# Description

### What is this workshop about? Provide as many details as possible. 

Satochip is an open-source hardware wallet based on a smartcard. The card is protected by a PIN code (4-16 chars) defined by the user: if a wrong PIN is entered 3 times in a row, the card will brick itself. The user can import a BIP39 seed into the card and the card will generate and protects the private keys through BIP32 key derivation. Once a seed is imported, the private keys are protected by the secure element and are never exported outside of the card. 

Being open-source and based on off-the-shelf hardware, Satochip makes it easy for anybody to build his own hardware wallet with high security and without having to trust any third party. It is also the most affordable hardware solutions, since smartcards are typically mass-produced and available from multiple manufacturers (NXP, Infineon, Gemalto to name a few). 

During this workshop, the attendees will:
* Learn what is a javacard and how it works
* Learn the security properties of smartcards
* Learn the different javacard models available 
* Learn the basics of javacard programming, using the Satochip applet as example
* Learn how to build (compile) the javacard code into a cap (binary) file using open source toolchain
* Learn how to load the applet firmware into the card
* Configure and use the resulting Satochip hardware wallet using a fork of Electrum 

The security model and security assumptions of Satochip will be discussed and the pros/cons of this approach will also be debated compared to other hardware solutions.
We are also currently working on the support of Schnorr signature and Taproot on Satochip, this can also be a topic of discussion.

Finally, the attendees will be able to bring their own hardware wallet back home. 

### What would an attendee learn from this workshop?

The attendees will learn frm A to Z how smartcards works and how they can be used to securely manage their bitcoin stash.

### Is there anything attendees should read up on before they attend this talk?

[Optional] introduction to javacard: https://www.oracle.com/java/technologies/java-card/javacard1.html
[optional] Satochip applet code available for review: https://github.com/Toporin/SatochipApplet
[optional] Electrum fork with Satochip plugin available for review: https://github.com/Toporin/electrum-satochip/tree/satochip/electrum/plugins/satochip

### Is there anything attendees should set up before the workshop?

Bring a laptop (Windows or Linux) with internet connection. 
We will provide the material needed to setup the hardware: a standard smartcard reader and the smartcard, but attendees can bring their own material if they wish.

### Relevant Links

# About the Speaker

I am currently the main developer at Satochip. Previously, I have also been working as security engineer and security auditor. I have a Phd in cryptography from the UCL (Université Catholique de Louvain) in Belgium, under the supervision of François-Xavier Standaert (expert in side-channel analysis) and Jean-Jacques Quisquater (cryptographer cited in the Bitcoin whitepaper). 
Linkedin: https://www.linkedin.com/in/baudoin-collard-74352510/

### Social Links
Github: https://github.com/Toporin
Twitter: https://twitter.com/satochipwallet?lang=en
Website: https://satochip.io
Telegram (support group): https://t.me/Satochip
email: info@satochip.io

# Workshop Details
### Length of workshop
2 hours (1 hour if we skip the introduction to javacard and load the firware into the card directly). 
It is also possible to split the workshop in two slots, with a 1 hour slot for reviewing/building the firmware, and 1 hour for loading/testing the card and discussion.

### Preferred Day/Time Slot 
no preferrence

<img src="https://user-images.githubusercontent.com/10826856/236769092-27e77b02-b5d1-4d8f-bf2e-b31c9ce0989c.jpg" width=50% height=50%>

**Comment by iglesiasbrandon** [comment]
@Toporin, thank you for submitting an issue for TABConf 2023! We will review all the issues as we get closer to the conference.

I don't expect us to add any issues to the schedule until a month or two before the conference but keep an eye on this in case someone asks questions. Also, please feel free to update the issue description if you'd like to add anything else as we get closer to the event.

**Comment by Toporin** [comment]
Hi @iglesiasbrandon ,

A quick message to reiterate our motivation to present the Satochip hardware wallet in a workshop!
By the way, we are currently working to integrate Satochip with Sparrow wallet, we will have a proof of concept ready for Tabconf, so this could be a nice complement to Craig Raw's workshop :-)

**Comment by miketwenty1** [comment]
@Toporin I don't believe this is going to be accepted. I would suggest if you want to help people hack/build you pick a spot at builders day and share about Satochip

**Comment by Toporin** [comment]
Thank @miketwenty1 for the notification! 
Living in good ol' Europe, I won't be able to make it this late, but I would really like to attend Tabconf next time and present our project. We will have many more improvements to show by then, including Taproot support and more wallet integrations! 
I wish everyone a great conference!

**Comment by miketwenty1** [comment]
@Toporin I hope to see next year! I appreciate the submission. One of our big criteria for accepting talks is reputation and community involvement around a topic. We also sort and order issues by 👍 emojis as one way to help understand how many people are excited to see a various talk/panel/workshop. Please consider submitting another idea next year. We'll be accepting talks for TABConf 6 in about 5-6 months.
